src/App.jsx

```diff
@@ -14,57 +15,82 @@ export const goods = [
   'Garlic',
 ];
 
-export const App = () => (
-  <main className="section container">
-    <h1 className="title is-flex is-align-items-center">No goods selected</h1>
+export const App = () => {
+  const [selectedGood, setSelectedGood] = useState('Jam');
+  const [buttonValue, setButtonValue] = useState('-');
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
-      <button data-cy="ClearButton" type="button" className="delete ml-3" />
-    </h1>
+  const checkButtonValue = good => {
+    if (good === selectedGood) {
+      setButtonValue(prev => {
+        if (prev === '-') {
+          setSelectedGood('');
+        }
+      });
```

Comment:
The `setButtonValue` function is called with a function that only updates the `selectedGood` state but does not update the `buttonValue` state. This means that the `buttonValue` will not change from `'-'` when the selected good is clicked again, which might not be the intended behavior. Consider updating the `buttonValue` state directly within this function.